tests/async/test_context_manager.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.async_api import BrowserType
+
+
+async def test_context_managers(browser_type: BrowserType, launch_arguments):
+    browser = await browser_type.launch(**launch_arguments)
+    async with browser:
```

Comment:
Is `async with variable` idiomatic? I thought that the whole point of context manager was to never leak the value outside of the context.

Reply:
Nope, the use of context manager is to handle the resource and do some exit work after done, not to leak the value etc